### PR TITLE
fix: dropdown property control

### LINKED ISSUE #10576: [Bug]: Decimal option “0” is not stored on the property pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currency Input Widget - The decimal option “0” is not stored on the property pane when the user chooses decimal option 1 or 2 and closing the property pane and again choosing the decimal value to 0 and on re-opening the property pane the value changes to 1

https://loom.com/share/96c7aad0dd72449e83795afd89bf8293

### Steps To Reproduce

1.Launch Appmsith app in release environment
2.Drag and drop currency input widget 
3.Open property pane and choose decimal option as 1 or 2
4.Click anywhere on canvas so that the property pane will close
5.Open property pane again and choose option as 0
6.Click anywhere on canvas so that the property pane will close
7.Open property pane again and observe the decimal value is 1 instead of 0

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

> Allow `0` value for Dropdown Property

Fixes #10576

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/dropdown-property-control 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.46 **(0)** | 37.06 **(0)** | 35.32 **(0)** | 55.85 **(0.01)**
 :green_circle: | app/client/src/components/propertyControls/DropDownControl.tsx | 88.89 **(0.43)** | 63.64 **(0)** | 71.43 **(0)** | 87.5 **(0.54)**</details>

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/currency_widget_spec.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const widgetName = "currencyinputwidget";
 describe("Currency widget - ", () => {
   before(() => {
     cy.addDsl(dsl);
+    cy.wait(7000);
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/currency_widget_spec.js:9]: 7 second wait time seems a bit execessive. What is the reason for such long wait time.

Comment (Tooluloope) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/FormWidgets/currency_widget_spec.js:9]: Oh, this is my hack to pass the test locally, forgot to remove it

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
Tested this PR and working as expected

**Pull request merged**